### ISSUE TITLE
Protocol Implementation Issues in Demo project

### ISSUE THREAD

**Issue by taimur** [devxoul/URLNavigator#49]
I downloaded the sample project and I faced this issue.

Type 'UserViewController' does not conform to protocol 'URLNavigable'. Here is the screen shot.
http://imgur.com/DDQxSjD

I researched and I found there is only 1 function in the Protocol URLNavigable

`init?(navigation: Navigation)`

and there is not function with this signature

` convenience init?(url: URLConvertible, values: [String: Any], userInfo: [AnyHashable: Any]?)`

 Using autocorrect this is what compiler added and with little modifications I am able to compile and run the project but its not navigating properly. Can someone help / assist to fix it.
Thanks a lot !
```
extension UserViewController: URLNavigable {
  convenience init?(navigation: Navigation) {
    guard let id = navigation.values["id"] as? Int else {
               return nil
        }
    self.init(navigation:navigation)
    
  }

  convenience init?(url: URLConvertible, values: [String: Any], userInfo: [AnyHashable: Any]?) {
    guard let username = values["username"] as? String else { return nil }
    self.init(username: username)
  }

}
```

**Comment by devxoul** [comment]
Oh I forgot to update the demo project code. It's because of the latest release ([1.2.0](https://github.com/devxoul/URLNavigator/releases/tag/1.2.0)) which has a breaking API changes. I'll update it soon. Thanks for reporting :)

**Comment by taimur** [comment]
Thanks a lot mate !

**Comment by devxoul** [comment]
Fixed on #50, thanks a lot @taimur!
